compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1914,6 +1914,16 @@ rustc_queries! {
         desc { "normalizing `{}`", goal.value.value }
     }
 
+    /// Do not call this query directly: invoke `normalize` instead.
+    query normalize_weak_ty(
```

Comment:
There are now three queries for very similar things. Should we just have a single query and dispatch via `tcx.def_kind(goal.def_id)` internally?

Reply:
I don't care really, we also have one query per MIR type op, so it's honestly better to just be explicit unless the code is particularly compelling to merge 😅